moped-etl/data-tracker-sync/data_tracker_sync.py

```diff
@@ -0,0 +1,198 @@
+        f"Updated Moped project {moped_project_id} with Knack ID {knack_project_id}"
+    )
+    return update
+
+
+def find_synced_moped_projects(last_run_date, is_test=False):
+    # If we are testing, request a Moped project record with a known ID.
+    synced_projects = []
+
+    if is_test:
+        data = make_hasura_request(
+            query=GET_TEST_SYNCED_PROJECTS,
+            variables={
+                "last_run_date": last_run_date,
+                "project_id": TEST_MOPED_PROJECT_ID,
+            },
+        )
+        synced_projects = data["moped_project"]
+    else:
+        data = make_hasura_request(
+            query=GET_SYNCED_PROJECTS,
+            variables={
+                "last_run_date": last_run_date,
+            },
+        )
+        synced_projects = data["moped_project"]
+
+    logger.info(f"Found {len(synced_projects)} synced projects")
+    logger.debug(f"Found synced projects: {synced_projects}")
+    return synced_projects
+
+
+def update_knack_project_from_moped_project(app, moped_project_record, is_test=False):
+    logger.debug(f"Updating Knack record for {moped_project_record}")
+
+    # Build Knack record and add existing Knack record ID for update
+    knack_project_record = build_knack_project_from_moped_project(
+        moped_project_record=moped_project_record, is_test=is_test
+    )
+    knack_project_record["id"] = moped_project_record["knack_project_id"]
+
+    updated = app.record(
+        method="update",
+        data=knack_project_record,
+        obj=KNACK_DATA_TRACKER_PROJECT_OBJECT,
+    )
+
+    knack_record_id = updated["id"]
+    return knack_record_id
+
+
+def main(args):
+    app = knackpy.App(
+        app_id=KNACK_DATA_TRACKER_APP_ID, api_key=KNACK_DATA_TRACKER_API_KEY
+    )
+
+    # Find all projects that are not synced to Data Tracker
+    unsynced_moped_projects = find_unsynced_moped_projects(is_test=args.test)
+
+    # Create a Knack project for each unsynced Moped project
+    created_knack_records = []
+    for project in unsynced_moped_projects:
+        moped_project_id = project["project_id"]
+        knack_record_id = create_knack_project_from_moped_project(
+            app=app, moped_project_record=project, is_test=args.test
+        )
+        created_knack_records.append(
+            {
+                "moped_project_id": moped_project_id,
+                "knack_record_id": knack_record_id,
+            }
+        )
+
+        # Update Moped project with Knack record ID of created record
+        update_moped_project_knack_id(moped_project_id, knack_record_id)
+
+    # Find all projects that are synced to Data Tracker to update them
+    synced_moped_projects = find_synced_moped_projects(
+        last_run_date=args.date, is_test=args.test
+    )
+
+    # Update synced Moped projects in Data Tracker and skip those just created
+    updated_knack_records = []
+    updates_to_skip = [record["moped_project_id"] for record in created_knack_records]
```

Comment:
We are checking for Moped project records that have been updated since the last successful Airflow DAG run. This script updates Moped project records itself since we need to add the Knack project record ID in a Moped project's `knack_project_id` column in order to mark it as synced. 

I still feel like this opens the door to updating Knack records with Moped data that hasn't actually been updated outside of setting the Knack project ID. I might need some rubber ducking here. 🤔

Reply:
i consider this an unrelated problem, but i've had mixed results with the activity logging. As you can see in this screenshot, the first activity event used a default behavior, while the second formatted the entry correctly:
<img width="1382" alt="Screenshot 2024-01-25 at 1 23 11 PM" src="https://github.com/cityofaustin/atd-moped/assets/14793120/2614c85f-ce8e-4d8e-8e53-12a48040115f">

i think i'm ok with continuing to have this event in the log, but we'll need ot fix that bugginess. i can open a separate issue for it.

Reply:
@johnclary this is reminding me of the random activity log entry that showed up as "updated is deleted" but I couldnt reproduce it

Reply:
I was stuck on the original question for a while, I didn't understand how there wasn't an infinite loop, but it finally clicked that the updated_at timestamp is only changed on the moped project here: `update_moped_project_knack_id` and not in the update_knack_project flow. 
I agree its not ideal, but seems better than having to update the trigger on every column renaming or addition. 
Will it be confusing in Data Tracker to see the records being updated twice, and a day apart?

Reply:
capturing the record sync date is an interesting idea, but i can't quite see how it all works together. if the sync date is always after the DAG run start date, and the `updated_at` date is always after both of those, i think we're still stuck without a way to know for sure if a record has changed since the last sync. but i'm not super confident about that 🤷

Reply:
I agree - it also sounds like it would make it more convoluted. I'm down to ship and see how this works out. Maybe something more solid will float our way! 🤞